public/src/pages/StudentPage.tsx

```diff
@@ -244,11 +247,11 @@ export class StudentPage extends ViewPage {
         return [];
     }
 
-    private onlyActiveCourses(studentCourse: IAssignmentLink[]): IAssignmentLink[] {
-        const userCourses: IAssignmentLink[] = [];
+    private onlyActiveCourses(studentCourse: IStudentLabsForCourse[]): IStudentLabsForCourse[] {
+        const userCourses: IStudentLabsForCourse[] = [];
         studentCourse.forEach((a) => {
-            if (a.link && (a.link.getStatus() === Enrollment.UserStatus.STUDENT
-                || a.link.getStatus() === Enrollment.UserStatus.TEACHER)) {
+            if (a.enrollment && (a.enrollment.getStatus() === Enrollment.UserStatus.STUDENT
```

Comment:
Here the same method for `isStudent` and `isTeacher` can be reused.

Reply:
I'm not sure adding methods to TS interfaces will make life easier, as those methods will have to be re-implemented at several places throughout the whole codebase (everywhere we currently can simply push an anonymous interface to an array).  
But it is also possible that I fail to see some better options. What if we just make it a separate task and I'll check all the options before adding more changes?

Reply:
Okay, I don't know enough TS to know this, but since this is a "closure" thing, couldn't there be a guard (instead of the if-stmt) to enter the anon function, which would then just be the `push(a)`. I will look for some examples of what I mean.